cli/debug.go

```diff
@@ -119,9 +124,73 @@ func runDebugRangeDescriptors(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+var debugRaftLogCmd = &cobra.Command{
+	Use:   "raft-log [directory] [range id]",
+	Short: "print the raft log for a range",
+	Long: `
+  Prints all log entries in a store for the given range.
+`,
+	RunE: runDebugRaftLog,
+}
+
+func printRaftLogEntry(kv engine.MVCCKeyValue) (bool, error) {
+	var meta engine.MVCCMetadata
+	if err := meta.Unmarshal(kv.Value); err != nil {
+		return false, err
+	}
+	value := roachpb.Value{
+		RawBytes: meta.RawBytes,
+	}
+	var ent raftpb.Entry
+	if err := value.GetProto(&ent); err != nil {
+		return false, err
+	}
+	if len(ent.Data) > 0 {
+		var cmd roachpb.RaftCommand
+		if err := cmd.Unmarshal(ent.Data[9:]); err != nil {
```

Comment:
From `storage.raftCommandIDLen`. I'll export `storage.decodeRaftCommand` and use it here instead.